diff --git a/bt_speaker.py b/bt_speaker.py
index 772dad6..0079da0 100755
--- a/bt_speaker.py
+++ b/bt_speaker.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/env python3
 
 from gi.repository import GLib
 from bt_manager.audio import SBCAudioSink
diff --git a/codecs/Makefile b/codecs/Makefile
index 7d9e623..4f6cfaf 100644
--- a/codecs/Makefile
+++ b/codecs/Makefile
@@ -1,5 +1,3 @@
-CC = gcc -O3
-CFLAGS =
 INSTALL_PREFIX = /usr
 HEADER = rtpsbc.h
 
@@ -8,10 +6,10 @@ OBJS = rtpsbc.o sbc.o sbc_primitives.o
 TARGET = librtpsbc.so
 
 %.o: %.c
-	$(CC) $(CFLAGS) -fPIC -c $< -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) -fPIC -c $< -o $@
 
 $(TARGET): $(OBJS)
-	$(CC) -shared $(OBJS) -o $(TARGET)
+	$(CC) -shared $(LDFLAGS) $(OBJS) -o $(TARGET)
 
 .PHONY: all clean
 
diff --git a/hooks.default/connect b/hooks.default/connect
index 1b99830..f122e92 100755
--- a/hooks.default/connect
+++ b/hooks.default/connect
@@ -1,6 +1,6 @@
-#!/bin/bash -x
+#!/bin/sh -x
 
-ogg123 /usr/share/sounds/freedesktop/stereo/service-login.oga
+#ogg123 /usr/share/sounds/freedesktop/stereo/service-login.oga
 
 # disconnect wifi to prevent dropouts - TODO: check what permissions bt-speaker might need for this (netdev group?)
-# ifconfig wlan0 down &
\ No newline at end of file
+# ifconfig wlan0 down &
diff --git a/hooks.default/disconnect b/hooks.default/disconnect
index 97d45e7..5f034df 100755
--- a/hooks.default/disconnect
+++ b/hooks.default/disconnect
@@ -1,6 +1,6 @@
-#!/bin/bash -x
+#!/bin/sh -x
 
-ogg123 /usr/share/sounds/freedesktop/stereo/service-logout.oga
+#ogg123 /usr/share/sounds/freedesktop/stereo/service-logout.oga
 
 # reenable wifi - TODO: check what permissions bt-speaker might need for this (netdev group?)
-# ifconfig wlan0 up &
\ No newline at end of file
+# ifconfig wlan0 up &
diff --git a/hooks.default/startup b/hooks.default/startup
index 5fda73c..989bc5d 100755
--- a/hooks.default/startup
+++ b/hooks.default/startup
@@ -1,4 +1,4 @@
-#!/bin/bash -x
+#!/bin/sh -x
 #
 # Commands to run after startup of bt_speaker
 
diff --git a/hooks.default/track b/hooks.default/track
index 4423c39..598c6f0 100755
--- a/hooks.default/track
+++ b/hooks.default/track
@@ -1,4 +1,4 @@
-#!/bin/bash -x
+#!/bin/sh -x
 
 # called if new track data is available
 # track data can be accessed from environment variables
@@ -8,4 +8,4 @@
 # echo Artist: $ARTIST
 # echo Album: $ALBUM
 # echo Genre: $GENRE
-# echo
\ No newline at end of file
+# echo
